tasks/discover-gateway-ip.yml

```diff
@@ -1,12 +1,27 @@
 ---
 
+- block:
+    - name: Query existing stack for its output
+      os_stack_outputs:
+        stack: "{{ cluster_name }}"
+      ignore_errors: true
+
+    - name: Extract output of existing ephemeral bastion
+      vars:
+        query: "node_groups[?name=='{{ cluster_gw_group }}'].nodes[0]"
+      set_fact:
+        fip_stack_output: "{{ openstack_stack_outputs | json_query(query) | first or omit }}"
```

Comment:
We should add `jmespath` to `requirements.txt` as it is required for `json_query`. It gets installed because it is a dependency of `openstacksdk`, but this could change in the future.

Reply:
Shouldn't it be caught as either a dependency of Ansible or of Jinja2, depending where the json_query filter lives? Not something we should have to worry about.

Reply:
I wish it was that simple: https://github.com/ansible/ansible/issues/24319#issuecomment-382023446

Reply:
Done